runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -584,6 +585,24 @@ private void hop(boolean previous)
 				}
 			}
 
+			// Check world region if filter is enabled
+			if (config.quickHopRegionFilter() != RegionFilterMode.NONE) {
+				switch (config.quickHopRegionFilter()) {
+					case AUSTRALIA:
+						if (world.getRegion() != WorldRegion.AUSTRALIA) continue;
+						break;
+					case GERMANY:
+						if (world.getRegion() != WorldRegion.GERMANY) continue;
+						break;
+					case UNITED_KINGDOM:
+						if (world.getRegion() != WorldRegion.UNITED_KINGDOM) continue;
+						break;
+					case UNITED_STATES:
```

Comment:
Maybe the switch can be eliminated by storing the WorldRegion mapping in RegionFilterMode. This way you can do something like ``world.getRegion() != config.quickHopRegionFilter().getRegion()`` to filter by world.

Reply:
That's something I had thought about doing but forgot to do. Anyway all your requests have been made now.